ecosystem/indexer/src/indexer/tailer.rs

```diff
@@ -9,12 +9,46 @@ use crate::{
 };
 use aptos_logger::info;
 use aptos_rest_client::Transaction;
+use serde_json::{json, Value};
 use std::{fmt::Debug, sync::Arc};
 use tokio::{sync::Mutex, task::JoinHandle};
 use url::{ParseError, Url};
 
 diesel_migrations::embed_migrations!();
 
+pub fn string_null_byte_replacement(value: &mut str) -> String {
+    value.replace('\u{0000}', "").replace("\\u0000", "")
+}
+
+pub fn recurse_remove_null_bytes_from_json(sub_json: &mut Value) {
+    match sub_json {
+        Value::Array(array) => {
+            for item in array {
+                recurse_remove_null_bytes_from_json(item);
+            }
+        }
+        Value::Object(object) => {
+            for (_key, value) in object {
+                recurse_remove_null_bytes_from_json(value);
+            }
+        }
+        Value::String(str) => {
+            if !str.is_empty() {
+                let replacement = string_null_byte_replacement(str);
+                *str = replacement;
+            }
+        }
+        _ => {}
+    }
+}
+
+pub fn remove_null_bytes_from_txn(txn: Arc<Transaction>) -> Arc<Transaction> {
+    let mut txn_json = json!(txn);
```

Comment:
can you do `serde_json::to_value(txn).unwrap();` ? Unsure all of what the `json!` macro is doing here

Reply:
I think it does the same thing no?

Reply:
Changed it